2020-01-Winter/0_arrays_sorting/Merge_Sorted_Arrays/mergeSortArr.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <vector>
+
+std::vector<int>& merge(std::vector<int>& nums1, int m, std::vector<int>& nums2, int n)
+{
+  int backOfArray = nums1.size()-1;
+
+  while(m-1 > 0)
+  {
+    if(nums1[m-1] < nums2[n-1])
+    {
+      nums1[backOfArray] = nums2[n-1];
+      std::cout << n;
+      n--;
+      backOfArray--;
+    }
+    else if (nums1[m-1] > nums2[n-1])
+    {
+      nums1[backOfArray] = nums1[m-1];
+      nums1[m-1] = 0;
+
+      m--;
+      backOfArray--;
+    }
+    else
+    {
+      nums1[backOfArray] = nums2[n-1];
+      n--;
+      m--;
```

Comment:
Be careful here; if you decrement both counters without adding the value twice, you will wind up with missing entries when the values would be added at a different index than they originate at.

Try it with this example: [0, 1, 1, 3] (length 10) and [1, 2, 2, 3, 4, 5]. You'll find that 0 is repeated twice and 3 is only present once.

I'd recommend only decrementing the nums2 counter; let the next iteration sort out nums1.

(duplicated from comment below)

```suggestion
```

Reply:
Yup... that's a bug.

When I removed the m-- it caused a lot of problems even with the given test. I've modified the function and it should now account for the case you've given above, as well as the case where nums1 runs out of stuff before nums2 does.

Added some additional tests to the .cpp file as requested.

Reply:
Thanks!! This is much cleaner :tada: